log/log.go

```diff
@@ -0,0 +1,14 @@
+package log
+
+// Logger is the least-common-denominator interface for all log operations.
+type Logger interface {
+	With(Field) Logger
```

Comment:
From my experience, the common case is to be providing many fields. Rarely providing a single field at a time. I'd fashion the API to serve that most common case better, maybe by having:

```go
With(...Fields) Logger
```